standards/mobile_app_standards.md

```diff
@@ -0,0 +1,135 @@
+
+Troubleshooting
+---------------
+
+Often, issues will only be encountered when running apps on actual target devices. **Therefore you must have the knowledge, kit, privileges and app distribution arrangements in place to be able to remotely debug the app on suitable devices.**
+
+Information about errors occurring on mobile devices will often be lost to us, making problem resolution much more difficult. To mitigate this issue **the app must log errors, and it must be possible for the user to make that information available to us for support purposes**, even if that is just via screenshot. It should also be possible to vary the level of detail logged, for example by implementing a "debug" mode. Early implementation of such features is very likely to also speed up error resolution during development itself.
+
+Extra Considerations for Testing Mobile Apps
+--------------------------------------------
+
+**Mobile Apps must be tested on a representative range of platforms and devices.** That range of platforms and devices will obviously depend on the intended use of the app: for example, it can include the Apple platform and associated devices for an internal app, but for an app intended for use beyond the DEFRA family will include at least both the Apple and Android platforms and associated devices.
+
+**At all times clearly state the specific operating system versions and device types on which the app will be tested, and where it is therefore "officially supported". This list will change constantly as operating systems and devices are updated.**
+
+Appropriate tools should be adopted to facilitate this. For example, if using a full functional IDE such as Xcode or Android Studio, make extensive use of the device simulators.
+
+However, no simulator is entirely perfect, and, once the mobile app has gone live, you will also be faced with the need to keep it tested and running despite rapid operating system updates (for example, for iPad OS these occur roughly monthly). It is therefore also strongly recommended that you acquire, and refresh as needed, a range of suitable devices for use in developing and then supporting the app.
+
+**If the app is to be deployed on DEFRA managed devices it must be tested on the currently latest DEFRA adopted version of iOS/iPad OS and the currently latest public version of iOS/iPad OS** (most of the time these versions will be the same, but there is normally a slight delay between Apple releasing a new version of iOS/iPad OS and DEFRA adoption, to ensure that no business critical apps have been negatively affected).
+
+**If the app is intended for use outside DEFRA, then it must be tested on the versions of iOS/iPad OS stated above, plus it should also be tested on the previous public version, at least. It must also be tested on the latest version of Android and should also be tested on the previous version, at least.**
```

Comment:
This seems likely to be highly contextual.

Android is a quagmire of versioning - it is highly unlikely that testing N-1 will be sufficient for a public user base.

I'd almost say that all bets are off with public Android apps and you just have to be ready to fix anything.

Reply:
Not sure how to turn that into an actual standard! Is the "at least" sufficient?